frontend/src/components/smart/PvPArenaSummary.vue

```diff
@@ -144,14 +157,61 @@ export default {
   },
 
   methods: {
+    ...mapActions([
+      'withdrawFromArena',
+      'getDuelQueue'
+    ]),
+
     setTab(tabNumber) {
       this.tab = tabNumber;
     },
 
+    handleErrorMessage(value, errorMessage, returnedMessage) {
+      if (value.includes(`reverted with reason string '${errorMessage}'`)) {
+        return this.$dialog.notify.error(returnedMessage);
+      }
+      return 'There has been an error. Try again.';
+    },
+
     async handleEnterArenaClick() {
       this.$emit('enterMatchMaking');
     },
+
+    async leaveArena() {
+      this.loading = true;
+
+      if (this.isCharacterInDuelQueue) {
+        alert(i18n.t('pvp.currentlyInDuel'));
+        this.loading = false;
+        return;
+      }
+
+      try {
+        await this.withdrawFromArena(this.currentCharacterId);
+
+        this.$emit('leaveArena');
+      } catch (err) {
+        console.log('leave arena error: ', err.message);
+
+        this.handleErrorMessage(err.message, 'Char not in arena', 'The character is not in the arena');
+        this.handleErrorMessage(err.message, 'Defender duel in process', 'Duel already in process');
+      }
+
+      this.loading = false;
+    },
   },
+
+  async created() {
+    this.loading = true;
+
+    this.duelQueue = await this.getDuelQueue();
+
+    if (this.duelQueue.includes(`${this.currentCharacterId}`)) {
+      this.isCharacterInDuelQueue = true;
+    }
+
+    this.loading = false;
```

Comment:
Same as before. In this function there's no `try...catch...finally` block, please add

Reply:
If that's what needed for the given scenario, sure thing, go for it. Overall just being stuck on the "loading" action is incorrect, when you need to try something again, let users... try again :D But I don't know the whole context behind it, for sure it will make users refresh the page